### PR TITLE
Redesign Timeline

### PR DESCRIPTION
**Pull request by brendanlaschke**
Redesign the tl to match figma (#1278). 

The other "new" activities are not implemented im leaving that for later (and maybe someone else :) ), therefore it may look a bit silly with only Notes and Tasks implemented ... 

Bildschirmfoto 2023-09-29 um 15 57 27

### CODE REVIEW

File: front/src/modules/activities/timeline/components/TimelingeActivityGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  padding-top: ${({ theme }) => theme.spacing(2)};
+  position: relative;
+`;
+
+const StyledActivityGroupBar = styled.div`
+  align-items: center;
+  background: ${({ theme }) => theme.background.secondary};
+  border: 1px solid ${({ theme }) => theme.border.color.light};
+  border-radius: 40px;
+  bototm: 0;
```
Comment (charlesBochet) [inline front/src/modules/activities/timeline/components/TimelingeActivityGroup.tsx:32]: typo + are these bottom, top, left properties required?

File: front/src/modules/activities/timeline/components/TimelingeActivityGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  padding-bottom: ${({ theme }) => theme.spacing(2)};
+  padding-top: ${({ theme }) => theme.spacing(2)};
+  position: relative;
+`;
+
+const StyledActivityGroupBar = styled.div`
+  align-items: center;
+  background: ${({ theme }) => theme.background.secondary};
+  border: 1px solid ${({ theme }) => theme.border.color.light};
+  border-radius: 40px;
```
Comment (charlesBochet) [inline front/src/modules/activities/timeline/components/TimelingeActivityGroup.tsx:31]: we should use a value in theme for border-radius

File: front/src/modules/activities/timeline/components/TimelingeActivityGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  gap: ${({ theme }) => theme.spacing(4)};
+`;
+const StyledMonthSeperatorLine = styled.div`
+  background: ${({ theme }) => theme.border.color.light};
+  border-radius: 50px;
+  flex: 1 0 0;
+  height: 1px;
+`;
+
+const monthNames = [
```
Comment (charlesBochet) [inline front/src/modules/activities/timeline/components/TimelingeActivityGroup.tsx:57]: is there a way, we don't need to hard core the months? I feel we should use native JS date api here, for example:
`const month = date.toLocaleString('default', { month: 'long' });`

**Comment by charlesBochet** [comment]
Thanks @brendanlaschke!  I've left comment on the code. Regarding the product behavior, we should make sure that the element fit in the page, without making it scrolling horizontically. We can either use %width for some element, or try a flex alternative. I'm unsure what's the best way to go here!

image

On mobile, the issue is particularly present:
image

File: front/src/modules/activities/timeline/components/TimelingeActivityGroup.tsx (hunk 1 above)
Comment (brendanlaschke) [reply front/src/modules/activities/timeline/components/TimelingeActivityGroup.tsx]: top required because of padding ...

**Comment by brendanlaschke** [comment]
Problem I still see is the mobile variant ... 
Maybe for this design it is too much text ? 

My thought would be for one line maybe:

-Avatar- created -Icon- "Some Note"

Or do you have a better idea how we could do this better for mobile and desktop? 

Current situation:
Bildschirmfoto 2023-09-30 um 18 37 41

**Comment by lucasbordeau** [comment]
Maybe @Bonapara will have a design fix for that ?

**Comment by Bonapara** [comment]
image

Here we go! 

Increase the spacing between lines from 12px to 24 px in mobile:

image

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=12021-35679&mode=design&t=OM6IvaXUAd2Idk3w-4

**Comment by brendanlaschke** [comment]
Great solution! Should be working now for mobile.

**Comment by charlesBochet** [comment]
@brendanlaschke sorry about the slow review, I will take a look tomorrow, I think we are very close to merge it @Bonapara and I will take a decision on the responsive issue

**Comment by lucasbordeau** [comment]
Waiting for activities refactor with new custom objects backend

**Review by lucasbordeau — APPROVED** [review]

**Pull request merged**